### PR TITLE
[SERVER-1894] Docs Update - Github secrets provisioning

### PR DESCRIPTION
**Pull request by circlery**
# Description
Updates docs to handle new github secrets provisioning methods.

# Reasons
server-1894

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: https://circleci.com/docs/style/.

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [ ] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

**Comment by atulsingh0** [comment]
@circlery It would be great if we can add "helm snippet" as well when the customer creates the secret. 
```
github:
  clientId:  ""
  clientSecret:  ""
```

and if enterprise

```
github:
  clientId:  ""
  clientSecret:  ""
  defaultToken: ""
```

**Review by lloydChris — CHANGES_REQUESTED** [review]
Can this match the structure that we have established above (https://github.com/circleci/circleci-docs/blob/4.0/server-preview/jekyll/_cci2/server/Installation/2-install-core.adoc#api-token) ?

**Comment by rosieyohannan** [comment]
Hello! Sorry I've made some changes to the filenames in the main branch for these changes so if you need help sorting out the conflicts let me know.

File: jekyll/_cci2/server/Installation/phase-2-core-services.adoc [hunk 1]
```diff
@@ -232,32 +232,67 @@ You will need to update your DNS records to the new loadbalancer once you have r
...
+[source,shell]
+----
+kubectl create secret generic github-secret \
+  --from-literal=clientId=<github-client-id> \
+  --from-literal=clientSecret=<github-client-secret> 
+----
+
+* *Option 2 - CircleCI creates the secret*
++
+You created your Github OAuth application in the prerequisite phase. Add the client ID and secret to
```
Comment (lloydChris) [inline jekyll/_cci2/server/Installation/phase-2-core-services.adoc:252]:
```suggestion
Add the client ID and secret to 
```

File: jekyll/_cci2/server/Installation/phase-2-core-services.adoc [hunk 2]
```diff
@@ -232,30 +232,78 @@ You will need to update your DNS records to the new loadbalancer once you have r
...
+=== GitHub
+These instructions are for the non-enterprise version of GitHub.
 
+* *Option 1 - Create the secret yourself*
++
+[source,shell]
+----
+kubectl create secret generic github-secret \
+  --from-literal=clientId=<github-client-id> \
+  --from-literal=clientSecret=<github-client-secret>
```
Comment (lloydChris) [inline jekyll/_cci2/server/Installation/phase-2-core-services.adoc:247]: I think it makes sense to be consistent with the naming, could go the other way too, i have not strong opinion on that

```suggestion
  --from-literal=clientId=<client-id> \
  --from-literal=clientSecret=<client-secret> 
```

File: jekyll/_cci2/server/Installation/phase-2-core-services.adoc [hunk 3]
```diff
@@ -232,30 +232,78 @@ You will need to update your DNS records to the new loadbalancer once you have r
...
+After creating the default token, there are two setup options:
+
+* *Option 1 - Create the secret yourself*
++
+[source,shell]
+----
+kubectl create secret generic github-secret \
+  --from-literal=clientId=<github-client-id> \
+  --from-literal=clientSecret=<github-client-secret> \
+  --from-literal=defaultToken=<github-default-token>
```
Comment (lloydChris) [inline jekyll/_cci2/server/Installation/phase-2-core-services.adoc:279]:
```suggestion
  --from-literal=clientId=<client-id> \
  --from-literal=clientSecret=<client-secret> \
  --from-literal=defaultToken=<default-token>
```

File: jekyll/_cci2/server/Installation/phase-2-core-services.adoc [hunk 4]
```diff
@@ -232,30 +232,78 @@ You will need to update your DNS records to the new loadbalancer once you have r
...
+[source,yaml]
+----
+github:
+  enterprise: true
+  hostname: <github-enterprise-hostname>
+----
 
+* *Option 2 - CircleCI creates the secret*
++
+You created your Github OAuth application in the prerequisite phase. Add `clientID`, `clientSecret` and `defaultToken` to
```
Comment (lloydChris) [inline jekyll/_cci2/server/Installation/phase-2-core-services.adoc:293]:
```suggestion
Add `clientID`, `clientSecret` and `defaultToken` to 
```

File: jekyll/_cci2/server/Installation/phase-2-core-services.adoc [hunk 5]
```diff
@@ -232,30 +232,78 @@ You will need to update your DNS records to the new loadbalancer once you have r
...
 ----
 
+
 [#github-enterprise]
 === GitHub enterprise
 
-In the case of GitHub enterprise add the following to the `GitHub` section.
+The instructions for GitHub enterprise are similar, with a few extra steps to enable enterprise and create the required default token.
+
+In GitHub, Create the `defaultToken` by navigating to **Settings > Developer Settings > Personal access tokens**. The default token requires no scopes.
```
Comment (lloydChris) [inline jekyll/_cci2/server/Installation/phase-2-core-services.adoc:268]:
```suggestion
In GitHub, Create the `defaultToken` by navigating to **Settings > Developer Settings > Personal access tokens**. The default token requires no scopes.  You will also need the client ID and secret you created with your Github OAuth application in the prerequisite phase.
```

**Review by lloydChris — APPROVED** [review]

File: jekyll/_cci2/server/Installation/phase-2-core-services.adoc [hunk 6]
```diff
@@ -232,32 +232,80 @@ You will need to update your DNS records to the new loadbalancer once you have r
...
 [source,yaml]
 ----
 github:
   clientId: <client-id>
   clientSecret: <client-secret>
 ----
 
+
 [#github-enterprise]
 === GitHub enterprise
```
Comment (ryanwohara) [inline jekyll/_cci2/server/Installation/phase-2-core-services.adoc:264]:
```suggestion
=== GitHub Enterprise
```

File: jekyll/_cci2/server/Installation/phase-2-core-services.adoc [hunk 7]
```diff
@@ -232,32 +232,80 @@ You will need to update your DNS records to the new loadbalancer once you have r
...
+  clientSecret: <client-secret>
   enterprise: true
   hostname: <github-enterprise-hostname>
   defaultToken: <token>
 ----
 
+
 [#object-storage]
-=== Object storage
+== Step Five: Object storage
```
Comment (ryanwohara) [inline jekyll/_cci2/server/Installation/phase-2-core-services.adoc:308]:
```suggestion
== Step five: Object storage
```

**Pull request merged**